### PR TITLE
[quant][improvement] Added quantized fill test for per channel quantized tensors

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78662
* __->__ #78661

Summary:
Previously, the quantization test suite only tested the fill operator
for per tensor quantized tensors. This PR adds a test case for per
channel quantized tensors.

The existing `test_qtensor_fill`, which the newly introduced
test function `test_qtensor_fill_per_channel` is based on, case had some ambiguous
naming conventions. This PR renames some of those variables to be
clearer.

Test Plan:
```
python test/test_quantization.py test_qtensor_fill_per_channel
```

Differential Revision: [D36823958](https://our.internmc.facebook.com/intern/diff/D36823958)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36823958).

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/core/test_quantized_tensor.py [hunk 1]
```diff
@@ -956,6 +956,38 @@ def test_qtensor_fill(self):
...
+        for qtype, fill_with, device in itertools.product(types, fills, devices):
+            scales = scales.to(device)
+            zero_points = zero_points.to(device)
+            ones = ones.to(device)
+            q_filled = torch._empty_per_channel_affine_quantized(
+                dims, scales=scales, zero_points=zero_points, device=device,
+                axis=axis, dtype=qtype)
+            q_filled.fill_(fill_with)
+            int_repr = torch.quantize_per_channel(ones * fill_with, scales=scales,
+                                                 zero_points=zero_points, axis=axis, dtype=qtype)
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_tensor.py:981]: +1 for lint

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36823958).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36823958).

**Pull request closed without merging**